dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -252,18 +257,26 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context)
             #region Update State
 
             // Update state with result
-            this.State.Update(result.Result.Trim());
+            this.State.Update(resultValue);
+
+            // Update Plan Result in State with matching outputs (if any)
+            if (this.Outputs.Intersect(step.Outputs).Any())
```

Comment:
I would suggest moving away of having a DefaultResultKey, since all the information is already in the plan.

Potentially I would add a `Results` property where you get the collection of results for the plan and you can actually do whatever you want with it, including Join.

Reply:
I'd really like to sticking with the State object and SKContext -- until those patterns change/evolve (SKRequest/SKResponse maybe?). Since Input/Result is overloaded in SKContext, having this special key is useful right now.